libandroid-navigation-ui/src/test/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModelTest.java

```diff
@@ -94,6 +98,19 @@ public void updateRoute_navigationIsNotUpdatedWhenChangingConfigurations() {
     verify(navigation, times(0)).startNavigation(route);
   }
 
+  @Test(expected = NullPointerException.class)
+  public void updateRoute_navigationErrorsOutWhenViewModelIsDestroyed() {
+    Application application = mock(Application.class);
+    MapboxNavigation navigation = mock(MapboxNavigation.class);
+    DirectionsRoute route = mock(DirectionsRoute.class);
+    MapConnectivityController mockedConnectivityController = mock(MapConnectivityController.class);
+    MapOfflineManager mapOfflineManager = mock(MapOfflineManager.class);
+    NavigationViewModel viewModel = new NavigationViewModel(application, navigation, mockedConnectivityController,
+            mapOfflineManager);
+    viewModel.onCleared();
+    viewModel.updateRoute(route);
+  }
+
```

Comment:
It'd be 💯 to add a test that verifies that `NavigationViewRouter#onDestroy` (`destroyRouter()`) is called when on clearing the view model.